lib/rex/proto/http/client.rb

```diff
@@ -31,6 +33,8 @@ def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, prox
     self.username = username
     self.password = password
     self.comm = comm
+    self.http_trace_proc_request = http_trace_proc_request
+    self.http_trace_proc_response = http_trace_proc_response
```

Comment:
It might make sense to rename these to be something more generic. i.e. `on_request` `on_response`, that will allow us to hook into this again in the future without being tied specifically to logging.

And similar to the other comments, as we're aiming towards having a single subscriber instance receive the on_request/on_response calls this will just be:

```ruby
self.subscriber = subscriber || HttpSubscriber.new
```

Note that the `HttpSubscriber` would be the default noop (no operation) implementation of an http subscriber which doesn't log anything by default. We can use that to ensure we don't need lots of `if subscriber.nil?` guard clauses in our implementation